### PR TITLE
Adds the Colormate

### PR DESCRIPTION
**Pull request by xPokee**
## About The Pull Request
Ports the colormate from https://github.com/fortune13-ss13/thewasteland/commit/9cc7fab017af536d7a338a9af20c575fd048daaf, a funky device that lets you recolor your clothes and gear without the fear of your paint washing off. Includes two modes, matrixed coloring and regular (think like a spray can)

## How This Contributes To The Towelstation Roleplay Experience
More customization

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  I tested it :)
</details>

## Changelog

:cl: xPokee, TripleShades
add: Adds the colormate
/:cl:

### CODE REVIEW

File: code/__HELPERS/icons.dm [hunk 1]
```diff
@@ -1468,3 +1468,19 @@ GLOBAL_LIST_EMPTY(transformation_animation_objects)
...
+	ASSERT(cm.len >= 9)
+	if(cm.len < 12) // fill in the rest
+		for(var/i in 1 to (12 - cm.len))
+			cm += 0
+	if(!islist(color))
+		color = ReadRGB(color)
+	color[1] = color[1] * cm[1] + color[2] * cm[2] + color[3] * cm[3] + cm[10] * 255
+	color[2] = color[1] * cm[4] + color[2] * cm[5] + color[3] * cm[6] + cm[11] * 255
+	color[3] = color[1] * cm[7] + color[2] * cm[8] + color[3] * cm[9] + cm[12] * 255
+	return rgb(color[1], color[2], color[3])
```
Comment (distributivgesetz) [inline code/__HELPERS/icons.dm:1486]: modularize, move it to a `modular_towelstation/master_code/<this file's path>` code file

File: code/__HELPERS/matrices.dm [hunk 2]
```diff
@@ -185,6 +185,12 @@ round(cos_inv_third+sqrt3_sin, 0.001), round(cos_inv_third-sqrt3_sin, 0.001), ro
 			output[offset+x] = round(A[offset+1]*B[x] + A[offset+2]*B[x+4] + A[offset+3]*B[x+8] + A[offset+4]*B[x+12]+(y == 5?B[x+16]:0), 0.001)
 	return output
 
+/**
+* Assembles a color matrix, defaulting to identity
+*/
+/proc/rgb_construct_color_matrix(rr = 1, rg, rb, gr, gg = 1, gb, br, bg, bb = 1, cr, cg, cb)
+	return list(rr, rg, rb, gr, gg, gb, br, bg, bb, cr, cg, cb)
+
```
Comment (distributivgesetz) [inline code/__HELPERS/matrices.dm:193]: modularize

File: code/game/objects/items/circuitboards/machines/machine_circuitboards.dm [hunk 3]
```diff
@@ -1519,3 +1519,12 @@
...
 	)
+///Towelstation Addition - COLORMATE
+/obj/item/circuitboard/machine/colormate
+	name = "Colormate"
+	greyscale_colors = CIRCUIT_COLOR_SERVICE
+	build_path = /obj/machinery/gear_painter
+	req_components = list(
+		/datum/stock_part/servo = 1,
+		/obj/item/stack/cable_coil = 2)
+///Towelstation addition end
```
Comment (distributivgesetz) [inline code/game/objects/items/circuitboards/machines/machine_circuitboards.dm:1530]: modularize, add to your clothesmate module directory

File: code/modules/research/designs/machine_designs.dm [hunk 4]
```diff
@@ -1096,3 +1096,14 @@
...
+//TOWELSTATION ADDITION - COLORMATE
+/datum/design/board/colormate
+	name = "ColorMate"
+	desc = "This machine lets you paint your gear."
+	id = "colormate"
+	build_path = /obj/item/circuitboard/machine/colormate
+	category = list (
+		RND_CATEGORY_MACHINE + RND_SUBCATEGORY_MACHINE_SERVICE
+	)
+	departmental_flags = DEPARTMENT_BITFLAG_SERVICE
```
Comment (distributivgesetz) [inline code/modules/research/designs/machine_designs.dm:1109]: modularize

**Review by distributivgesetz — CHANGES_REQUESTED** [review]
if you ported this from somewhere, please credit it. we don't plagiarize

**Comment by xPokee** [comment]
> if you ported this from somewhere, please credit it, in this PR and the module's readme. we don't plagiarize

I'll try to track it down. It's a port of a port of probably yet another port

**Review by distributivgesetz — CHANGES_REQUESTED** [review]
doesn't look like you included the new master files in the .dme, looks good otherwise

**Review by distributivgesetz — APPROVED** [review]

**Pull request merged**